### PR TITLE
Error checking for cli exec-commands

### PR DESCRIPTION
**Pull request by shorsher**
faas-cli will now check the exit status of exec commands
  (i.e. <docker build ...>) and will "stop the world" if
  a non-zero exit occurs.

Signed-off-by: Alex Shorsher <alex.shorsher@gmail.com>

## Description

Previously, even if `faas-cli build` failed when building a container from a Dockerfile, faas-cli would continue executing.  The correct behavior should be to stop faas-cli execution, log, and present the user with a noticeable error message.  

The issue originally just referenced throwing an error if `faas-cli build` fails, however, it seems my changes would also add error checking for other commands that use ExecuteCommand() (currently just `faas-cli push`).  I think this is acceptable?

## Motivation and Context

#138 

## How Has This Been Tested?

Here are some gif's of me running tests using the included samples.  They include a single function building  w/ a Dockerfile failing as expected, multiple functions with an expected failing Dockerfile, and one correct build using the samples.  Apologizing in advance for the crap gif quality.  

**Single function failure**
http://gph.is/2g98SlH
**Multi-function failure**
http://gph.is/2g9nmlx
**Passing example**
http://gph.is/2g7SZfs

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: builder/exec.go [hunk 1]
```diff
@@ -4,8 +4,12 @@
 package builder
 
 import (
+	"fmt"
+	"log"
 	"os"
 	"os/exec"
+
+	"github.com/morikuni/aec"
```
Comment (johnmccabe) [inline builder/exec.go]: What OS/Terminals have you validated this on? Mainly have you checked what it looks like on Windows Cmd and Powershell?

Comment (alexellis) [reply builder/exec.go]: I added `aec` for version/new commands which rendered on Windows cmd + PowerShell on Windows 10. Would be good to have it checked on one of your VMs if you have time.

Comment (shorsher) [reply builder/exec.go]: I’ve only tested in macOS in iTerm and Terminal. I don’t have any Linux vm’s on hand but can spin one up if necessary.

**Comment by shorsher** [comment]
Tested aec coloring on Ubuntu 16.04, here's an attached screenshot.  
screen shot 2017-10-04 at 11 53 17 pm

**Comment by alexellis** [comment]
Thanks. Colouring works well on Bash. It tends to be Windows that can have problems we already have similar code in the CLI so we need to see how it is there. If you don't have access we can see if someone else does in the channel.

**Comment by alexellis** [comment]
Will take this change as it's better than what we had - but I think the error checking needs to be higher up in the code - so we are perhaps just setting the nonzero exit code and printing a message. In retrospect log.Fatal might be better after the other functions have had a chance to build.

**Comment by shorsher** [comment]
I can get started on v2 and submit another PR

**Comment by alexellis** [comment]
Great

**Comment by alexellis** [comment]
Thank you

**Pull request merged**